org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -4635,4 +4640,97 @@ private boolean getSplitSwitchLabelProposal(IInvocationContext context, ASTNode
 
 		return true;
 	}
+	private boolean getConvertFieldNamingConventionProposal(IInvocationContext context, ASTNode node, Collection<ICommandAccess> resultingCollections) {
+		if (!(node instanceof SimpleName)) {
+			return false;
+		}
+		SimpleName name= (SimpleName) node;
+		String selectedField = name.toString();
+		int offset = node.getStartPosition();
+		int length = node.getLength();
+		if (name.getAST().apiLevel() >= ASTHelper.JLS10 && name.isVar()) {
+			return false;
+		}
+
+		IEditorPart editor= ((AssistContext) context).getEditor();
+		if (!(editor instanceof JavaEditor))
+			return false;
+
+        @SuppressWarnings("unused")
+		List<String> fieldList = new ArrayList<String>();
+		CompilationUnit astRoot= context.getASTRoot();
+        TypeDeclaration typeDec = (TypeDeclaration) astRoot.types().get(0);
+        for(FieldDeclaration fd:typeDec.getFields()) {
+        	VariableDeclarationFragment fragment= (VariableDeclarationFragment) fd.fragments().get(0);
+			String fieldName= fragment.getName().getIdentifier();
+			int modifier = fd.getModifiers();
+			if(fieldName.equals(name.toString())) {
+				if(modifier!=24 && modifier!=25 && modifier!=26 && modifier!=28) {
+					return false;
+					}
+				if(isValidConstantName(fieldName)) {
+					return false;
+				}
+			}
+			fieldList.add(fieldName);
+        }
+        if(fieldList.size()<1 || !fieldList.contains(name.toString())) {
+        	return false;
+        }
+
+		if (resultingCollections == null) {
+			return true;
+		}
+
+		for(FieldDeclaration fd:typeDec.getFields()) {
+			VariableDeclarationFragment fragment= (VariableDeclarationFragment) fd.fragments().get(0);
+			String fieldName= fragment.getName().getIdentifier();
+			int fieldModifier = fd.getModifiers();
+			if(fieldName.equals(selectedField) && isValidConstantName(fieldName) == false && (fieldModifier==24 || fieldModifier==25 || fieldModifier==26 || fieldModifier==28)) {
+				String newName= convertToConstantName(fieldName);
+				if (!newName.equals(fieldName)) {
+					Image image= JavaPluginImages.get(JavaPluginImages.IMG_CORRECTION_RENAME);
+					ConvertFieldNamingConventionProposal proposal= new ConvertFieldNamingConventionProposal(newName, offset, length, offset, image, fieldName, context);
+					proposal.setCommandId(FIELD_NAMING_CONVENTION);
+					resultingCollections.add(proposal);
+				}
+			}
+		}
+		return true;
+	}
+	private static boolean isValidConstantName(String name) {
+	        char[] chars = name.toCharArray();
+	        boolean lowercaseEncountered = false;
+	        for (int i = 0; i < chars.length; i++) {
+	            char currentChar = chars[i];
+	            if (Character.isUpperCase(currentChar)) {
```

Comment:
Setting lowercaseEncountered to true doesn't make sense when you are testing for isUpperCase().  Can you not simply look for a lower-case character via isLowerCase() and return false if found?